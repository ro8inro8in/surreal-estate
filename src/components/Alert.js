import React from "react";
import "../styles/Alert.css";
import PropTypes from "prop-types";

const Alert = ({ message, success }) => {
  return (
    <div className={`Alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};
Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};
Alert.defaultProps = {
  success: false,
};
export default Alert;
