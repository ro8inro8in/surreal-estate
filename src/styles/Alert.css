.alert-error {
    display: flex;
    justify-content: center;
    padding: 5%;
    margin: 5%;
    background-color: rgb(255, 16, 16);
    color: white;
    border-radius: 12px;
    box-shadow: 0 8px 16px rgb(38 38 48 / 20%);
    box-sizing: border-box;
    font-size: 12px;
    
  }

  .alert-success {
    display: flex;
    justify-content: center;
    padding: 5%;
    margin: 5%;
    background-color: rgba(131, 0, 131, 0.753);
    color: white;
    border-radius: 12px;
    box-shadow: 0 8px 16px rgb(38 38 48 / 20%);
    box-sizing: border-box;
    font-size: 12px;
  }


