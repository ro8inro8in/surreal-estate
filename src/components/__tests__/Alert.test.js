import React from "react";
import { render } from "@testing-library/react";
import Alert from "../Alert";

test("displays an error message", () => {
  const { getByText } = render(<Alert message="Error!" />);

  expect(getByText(/Error/).textContent).toBe("Error!");
});

test("displays an success message", () => {
  const { getByText } = render(<Alert message="Success!!!" success />);

  expect(getByText(/Success/).textContent).toBe("Success!!!");
});

//need to check over theses tests this evening 
test("does not render an error or a success message if message props is empty", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);
  const alert = asFragment();
  expect(alert).toMatchSnapshot();
  expect(getByText(/Error/)).toMatchSnapshot("Error!");
  expect(asFragment()).toMatchSnapshot();
});
