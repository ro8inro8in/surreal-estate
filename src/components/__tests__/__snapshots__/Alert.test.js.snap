// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error or a success message if message props is empty 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`does not render an error or a success message if message props is empty 2`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`does not render an error or a success message if message props is empty: Error! 1`] = `
<div
  class="Alert alert-error"
>
  Error!
</div>
`;
